src/web/components/ClientSideTokenGeneration/CstgDomainsTable.tsx

```diff
@@ -0,0 +1,216 @@
+        </button>
+        <button
+          type='button'
+          className='transparent-button'
+          onClick={() => setOpenConfirmation(false)}
+        >
+          Cancel
+        </button>
+      </div>
+    </Dialog>
+  );
+}
+
+export type CstgDomainsTableProps = {
+  domains: string[];
+  onUpdateDomains: (domains: string[]) => Promise<void>;
+};
+
+export function CstgDomainsTable({ domains, onUpdateDomains }: CstgDomainsTableProps) {
+  const [showNewRow, setShowNewRow] = useState<boolean>(false);
+  const [selectedDomains, setSelectedDomains] = useState<string[]>([]);
+  const isSelectedAll = domains.length && domains.every((d) => selectedDomains.includes(d));
+  const getCheckboxStatus = () => {
+    if (isSelectedAll) {
+      return TriStateCheckboxState.checked;
+    }
+    if (selectedDomains.length > 0) {
+      return TriStateCheckboxState.indeterminate as CheckedState;
+    }
+    return TriStateCheckboxState.unchecked;
+  };
+
+  const checkboxStatus = getCheckboxStatus();
+
+  const handleCheckboxChange = () => {
+    if (checkboxStatus === TriStateCheckboxState.unchecked) {
+      setSelectedDomains(domains);
+    } else {
+      setSelectedDomains([]);
+    }
+  };
+  const isDomainSelected = (domain: string) => selectedDomains.includes(domain);
+
+  const handleBulkDeleteDomains = (deleteDomains: string[]) => {
+    onUpdateDomains(domains.filter((domain) => deleteDomains.includes(domain)));
+  };
+
+  const handleSelectDomain = (domain: string) => {
+    if (isDomainSelected(domain)) {
+      setSelectedDomains(selectedDomains.filter((d) => d !== domain));
+    } else {
+      setSelectedDomains([...selectedDomains, domain]);
+    }
+  };
+
+  const onAddRow = () => {
+    setShowNewRow(true);
+  };
+
+  const handleAddNewDomain = async (newDomain: string) => {
+    await onUpdateDomains([...domains, newDomain]);
+    setShowNewRow(false);
+  };
+
+  return (
+    <div className='cstg-domains-management'>
+      <h2>Top-level Domains</h2>
+      <div className='table-actions'>
+        <TriStateCheckbox onClick={handleCheckboxChange} status={checkboxStatus} />
+        {selectedDomains.length > 0 && (
+          <DeleteDomainDialog
+            onDeleteDomains={() => handleBulkDeleteDomains(selectedDomains)}
+            selectedDomains={selectedDomains}
+          />
+        )}
+        <button
+          className='transparent-button table-action-button'
+          type='button'
+          disabled={showNewRow}
+          onClick={onAddRow}
+        >
+          <FontAwesomeIcon icon='plus' className='cstg-domains-management-icon' />
+          Add Domain
+        </button>
+      </div>
+      <table className='cstg-domains-table'>
+        <thead>
+          <tr>
+            <th> </th>
+            <th className='domain'>Domain</th>
+            <th className='action'>Actions</th>
+          </tr>
+        </thead>
+        <tbody>
+          {domains.map((domain) => (
+            <CstgDomainItem
```

Comment:
Do we plan to have existing domain items as un-editable? So only add and delete?

Reply:
We don't really have an ID on domains, so I am sure if we need un-editable or not.

Reply:
yeah I guess it's just a string. seems reasonable